pkg/external_repos/introspect.go

```diff
@@ -107,76 +129,6 @@ func Introspect(ctx context.Context, repo *dao.Repository, dao *dao.DaoRegistry)
 	return total, nil, true
 }
 
-func reposForIntrospection(urls *[]string, force bool) ([]dao.Repository, []error) {
-	repoDao := dao.GetRepositoryDao(db.DB)
-	ignoredFailed := !force // when forcing introspection, include repositories over FailedIntrospectionsLimit
-	if urls != nil {
-		var repos []dao.Repository
-		var errors []error
-		for i := 0; i < len(*urls); i++ {
-			repo, err := repoDao.FetchForUrl((*urls)[i])
-			if err != nil {
-				errors = append(errors, err)
-			} else if ignoredFailed && repo.FailedIntrospectionsCount > config.FailedIntrospectionsLimit {
-				continue
-			} else {
-				repos = append(repos, repo)
-			}
-		}
-		return repos, errors
-	} else {
-		repos, err := repoDao.List(ignoredFailed)
-		return repos, []error{err}
-	}
-}
-
-// IntrospectAll introspects all repositories
-// Returns the number of new RPMs inserted system-wide, all non-fatal introspection errors,
-// and separately all other fatal errors
-func IntrospectAll(ctx context.Context, urls *[]string, force bool) (int64, []error, []error) {
-	var (
-		total                  int64
-		count                  int64
-		err                    error
-		dao                    = dao.GetDaoRegistry(db.DB)
-		introspectionErrors    []error
-		introspectFailedUuids  []string
-		introspectSuccessUuids []string
-		updated                bool
-	)
-	repos, errors := reposForIntrospection(urls, force)
-	for i := 0; i < len(repos); i++ {
-		if !force {
-			hasToIntrospect, reason := needsIntrospect(&repos[i])
-			log.Info().Msg(reason)
-			if !hasToIntrospect {
-				continue
-			}
-		} else {
-			log.Info().Msgf("Forcing introspection for '%s'", repos[i].URL)
-		}
-		count, err, updated = Introspect(ctx, &repos[i], dao)
-		total += count
-
-		if err != nil {
-			introspectionErrors = append(introspectionErrors, fmt.Errorf("Error introspecting %s: %s", repos[i].URL, err.Error()))
-			introspectFailedUuids = append(introspectFailedUuids, repos[i].UUID)
-		} else if updated {
-			introspectSuccessUuids = append(introspectSuccessUuids, repos[i].UUID)
-		}
-
-		err = UpdateIntrospectionStatusMetadata(repos[i], dao, count, err)
-		if err != nil {
-			errors = append(errors, err)
-		}
-	}
-
-	// Logic to handle notifications
-	sendIntrospectionNotifications(introspectSuccessUuids, introspectFailedUuids, dao)
-
-	return total, introspectionErrors, errors
-}
```

Comment:
There isn't really any use for a 'All' function here.   Its always done on a per repo basis via  task, so i removed it.